moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/jog_cpp_interface.h

```diff
@@ -95,5 +101,6 @@ class JogCppInterface : public JogInterfaceBase
   ros::NodeHandle nh_;
 
   std::atomic<bool> stop_requested_;
+  std::atomic<bool> paused_;
```

Comment:
This should probably not exist if the information already exists in `jog_calcs_`. Instead I would add a get method to jog_calcs.h for getting `pause_outgoing_jog_cmds_`. That way you never need to worry about keeping things in sync

Reply:
I added these (stop_requested and paused) to the shared_variables so that all threads use the same atomic bools